Syslog  added to documentation

KellerLiptrap opened the pull request:
Addition of Chasten Log command documentation to the read me.

simojo commented on README.md:
Hi @KellerLiptrap, after just a quick look I was wondering, what is the purpose of the shell code block here? It might be helpful to put what command produces this output. Additionally, because it's not shell syntax and rather the output, you may want to remove the `shell` keyword from the top of the code block. I'm only saying this because I see that the word `for` is syntax highlighted when that's not the desired behavior.

gkapfham commented:
Hello @KellerLiptrap can you please write a description for this PR?

simojo commented:
When you run `chasten log`, chasten opens up a websocket server on `127.0.0.1:2525` (See `chasten/server.py`). This then implements a handler callback that prints messages to the console and logs them to the file `.discover.log`:

```python
# server.py
class SyslogUDPHandler(socketserver.BaseRequestHandler):
    """Syslog UDP handler for receiving debugging messages."""

    def handle(self):
        """Receive a message and then display it in output and log it to a file."""
        global logger  # noqa: disable=PLW0603
        # receive the message from the syslog logging client
        message = bytes.decode(
            self.request[0].strip(), encoding=constants.server.Utf8_Encoding
        )
        # remove not-printable characters that can appear in message
        enhanced_message = str(message).replace(
            constants.markers.Bad_Fifteen, constants.markers.Empty_String
        )
        enhanced_message = enhanced_message.replace(
            constants.markers.Bad_Zero_Zero, constants.markers.Empty_String
        )
        # display the message inside of the syslog's console
        output.console.print(enhanced_message)
        # write the logging message to a file using a rotating file handler
        logger.debug(enhanced_message)
```

You can learn more about this by reading the source code.

I'm not completely sure about the whole intended functionality of this, but I would assume that, depending on the log level specified, the separate syslog process would pick up on this. Does anyone else have any more ideas as to how this works? I've only looked at it for a few minutes.

boulais01 commented:
@KellerLiptrap Did you resolve the issue that Simon pointed out? If so, we can check that as resolved and add this to the list of PRs we can merge by Tuesday.

KellerLiptrap commented:
@boulais01 Yes I have resolved these issues and updated this issue and added to the documentation. I believe that it can be added to the list to be merged.

hayleepierce commented:
@KellerLiptrap If this PR is ready for review, make sure to add the label `ready-for-review`.

laurennevill commented:
@gkapfham, this is priority 4

jnormile commented on README.md:
```diff
@@ -275,6 +275,38 @@ create the tool's command-line arguments and options through a terminal user
 `chasten` you can type the command `chasten interact`.
 
+## 📊Log
+`Chasten` has a built-in System Log. While using chasten you can use the command `chasten log` in your terminal. The system log feature allows the user to see events and messages that are produced by `chasten`. In addition, the `chasten log` feature will assist in finding bugs and the events that led to the bug happening. For the `chasten` program to display to the system log you will have to ope ...
```
Super nitpicky, but given that we want our documentation to also be "production ready", it's warranted. Whenever you use the phrase "system log", please be _consistent_ with capitalization (System Log vs. system log).

laurennevill commented:
Hello @KellerLiptrap, can you ensure that this PR is up to date with the main branch? From there we can get this merged

simojo commented on README.md:
```diff
@@ -275,6 +275,38 @@ create the tool's command-line arguments and options through a terminal user
+`Chasten` has a built-in system log. While using chasten you can use the command `chasten log` in your terminal. The system log feature allows the user to see events and messages that are produced by `chasten`. In addition, the `chasten log` feature will assist in finding bugs and the events that led to the bug happening. For the `chasten` program to display to the system log you will have to ope ...
+
+For example `chasten datasette-serve --debug-level DEBUG --debug-dest SYSLOG 
+< database path to file>` with produce the following output in the system log.
```
Please change "with" to "will"

boulais01 commented:
@laurennevill @jnormile I believe the issues you both brought up should be resolved now, if you wouldn't mind reviewing again.

tuduun reviewed (approved):
Good job on getting the requests done!

tuduun reviewed (approved):
Approved the Capitalization

boulais01 commented:
@laurennevill this is ready to be merged when you deem it fit.
